google/cloud/testing_util/integration_test.cc

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/testing_util/integration_test.h"
+#include "google/cloud/log.h"
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace testing_util {
+
+void IntegrationTest::TearDown() {
+  if (HasFailure()) ::google::cloud::LogSink::Instance().Flush();
```

Comment:
Googletest allows setup/teardown logic in constructors and destructors, which is nice because we don't have to worry about subclasses remember to call the super class's function at the right point. If the logic that's expected to go into this `IntegrationTest` fixture would work in its c'tor or d'tor, consider using those instead. Of course, there are some technical differences that sometimes requires SetUp/TearDown, in which case, of course use the one that works :-)

https://github.com/google/googletest/blob/master/docs/faq.md#should-i-use-the-constructordestructor-of-the-test-fixture-or-setupteardown-ctorvssetup

Reply:
I like that better.